indexer/indexer_tests/editable_map_object_test.cpp

```diff
@@ -289,4 +289,461 @@ UNIT_TEST(EditableMapObject_SetInternet)
   setInternetAndCheck(bunkerEmo, osm::Internet::Wlan, true);
   setInternetAndCheck(bunkerEmo, osm::Internet::Wlan, true);
 }
+
+UNIT_TEST(EditableMapObject_RemoveFakeNames)
+{
+  EditableMapObject emo;
+  StringUtf8Multilang name;
+  osm::FakeNames fakeNames;
+
+  name.AddString(GetLangCode("default"), "Default name");
+  name.AddString(GetLangCode("ru"), "Default name");
+  name.AddString(GetLangCode("en"), "Default name");
+  fakeNames.names.push_back({GetLangCode("ru"), "Default name"});
+  fakeNames.names.push_back({GetLangCode("en"), "Default name"});
+  fakeNames.defaultName = "Default name";
+
+  EditableMapObject::RemoveFakeNames(fakeNames, name);
+
+  size_t counter = 0;
+  name.ForEach([&counter](int8_t const code, string const & name) {
+    TEST_EQUAL(name, "Default name", ());
+    ++counter;
+    return true;
+  });
+  TEST_EQUAL(counter, 1, ());
+
+  name.Clear();
+  fakeNames.names.clear();
+  fakeNames.defaultName.clear();
+
+  name.AddString(GetLangCode("default"), "Default name");
+  name.AddString(GetLangCode("ru"), "Changed name");
+  name.AddString(GetLangCode("en"), "Default name");
+  fakeNames.names.push_back({GetLangCode("ru"), "Default name"});
+  fakeNames.names.push_back({GetLangCode("en"), "Default name"});
+  fakeNames.defaultName = "Default name";
+
+  EditableMapObject::RemoveFakeNames(fakeNames, name);
+
+  counter = 0;
+  name.ForEach([&counter](int8_t const code, string const & name) {
+    if (code == StringUtf8Multilang::kDefaultCode)
+      TEST_EQUAL(name, "Default name", ());
+    else if (code == GetLangCode("ru"))
+      TEST_EQUAL(name, "Changed name", ());
+    else
+      TEST(false, ());
+    ++counter;
+    return true;
+  });
+
+  TEST_EQUAL(counter, 2, ());
```

Comment:
StringUtf8Multilang can't contain duplicate names, therefore counter is not needed.